insights.go

```diff
@@ -21,29 +18,19 @@ func unregisterInsights() error {
 }
 
 func insightsIsRegistered() (bool, error) {
-	var errBuffer bytes.Buffer
-	cmd := exec.Command("/usr/bin/insights-client", "--status")
-	cmd.Stderr = &errBuffer
-
-	err := cmd.Run()
-
+	// While `insights-client --status` properly checks for registration status by
+	// asking Inventory, its two modes (legacy v. non-legacy API) behave
+	// differently (they return different texts with different exit codes).
+	// The `.registered` file is always present on the registered system; the worst
+	// thing that may happen is that we're out of date and the insights-client does
+	// not yet know it should unregister. However, this detection is reliable
+	// enough for our reporting use-case.
+	_, err := os.Stat("/etc/insights-client/.registered")
```

Comment:
Hidden file in `/etc` directory... What a bright idea to use such file for any purpose. Why do we have e.g. POSIX?

This is also horrible, but it is less painful than parsing output of command. It is also mush faster than running `insights-client --status`.

Unfortunately, it looks like that it is not reliable. Or how should I understand to this:

```
[root@rhel9 ~]# test -f /etc/insights-client/.registered 
[root@rhel9 ~]# echo $?
0
[root@rhel9 ~]# insights-client --status
System is registered locally via .registered file. Registered at 2024-08-22T11:03:02.414759
Insights API says this machine is NOT registered.
System unregistered locally via .unregistered file
[root@rhel9 ~]# echo $?
1
```

Actually, why do I see it? I don't aware that I have done anything unusual with this system. The system is registered (consumer certificate/key is installed and it is valid, not deleted on server side).

Reply:
It's an unfortunate precedent that has been set by previous authors of `insights-client`. We can't really remove it until we have an opportunity to break CLI compatibility, so we're stuck with it for now. The way the marker files are intended to work is: if `/etc/insights-client/.registered` exists, the client believes it is registered with console.redhat.com. If `/etc/insights-client/.unregistered` exists, the client believes it is not registered. There are probably cases that can come about where those marker files get improperly set, but it's the best we've got besides parsing string output.

Reply:
Does running `insights-client --status` alter the .registered/.unregistered marker files at all?

Reply:
It doesn't. It may only speed up the detection of the host being gone from Inventory -- if we want to ensure we have the most up to date information, we could run status. But I don't think that's necessary.

Reply:
Wait. As you can see here:

```
[root@rhel9 ~]# test -f /etc/insights-client/.registered 
[root@rhel9 ~]# echo $?
0
[root@rhel9 ~]# insights-client --status
System is registered locally via .registered file. Registered at 2024-08-22T11:03:02.414759
Insights API says this machine is NOT registered.
System unregistered locally via .unregistered file
[root@rhel9 ~]# echo $?
1
```

running `insights-client --status` deleted `/etc/insights-client/.registered` and created `/etc/insights-client/.unregistered`. So, it can alter these files.

Is it any corner case and we can ignore it?

Reply:
How did that `.registered` file get there, via proper client registration? If so, was it recently? If you haven't run the client in a long time (e.g. the VM was off), it might have been removed from Inventory after a staleness period, so it would make sense to unregister on status.

Running `--status` forces the client to check with Inventory. Regularly that's not a problem, because the system checks in once a day and the registration status is up to date, and this change (deletion of `.registered` and creation of `.unregistered`) would be done during a night.

Reply:
That being said, we can run `--status` before detecting the presence of the file. I've updated the PR.